### PR TITLE
Issue/8008

### PR DESCRIPTION
**Pull request by ilci**

### CODE REVIEW

File: config/sync/block.block.subtheme_videogame_help.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 uuid: e2ab0fbd-3718-4460-8c34-403e2a0b1dcb
```
Comment (dzadza) [inline config/sync/block.block.subtheme_videogame_help.yml:1]: If this block is not necessary, you can also delete it.

File: config/sync/core.entity_form_display.block_content.genres_to_play.default.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
+uuid: 129f1831-1e93-4a27-9225-3f745a2a732a
+langcode: en
+status: true
+dependencies:
+  config:
+    - block_content.type.genres_to_play
+    - field.field.block_content.genres_to_play.body
+    - field.field.block_content.genres_to_play.field_genre1
```
Comment (dzadza) [inline config/sync/core.entity_form_display.block_content.genres_to_play.default.yml]: Instead of having 7 fields, you can use only one multivalue field here.

File: config/sync/editor.editor.full_html.yml [hunk 3]
```diff
@@ -45,19 +45,22 @@ settings:
...
           name: Tools
           items:
             - ShowBlocks
             - Source
   plugins:
+    language:
+      language_list: un
     stylescombo:
-      styles: ''
+      styles: "h1.heading1|Heading 1\r\nh2.heading2|Heading 2\r\nh3.heading3|Heading 3\r\np.parbold|Paragraph bold\r\nspan.orangeupper|Orange uppercase text\r\nbutton.btn.btn-primary|Button primary\r\nbutton.btn.btn-secondary|Button secondary"
```
Comment (dzadza) [inline config/sync/editor.editor.full_html.yml]: I'm just thinking that instead of the `button` element, use can the `anchor` element as well to create buttons. Is there a `button` element selectable from the editor?

File: web/themes/custom/subtheme_videogame/src/components/field/_companyinformation.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.companyinformation {
+  margin: 5% 10% 5% 10%;
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/components/field/_companyinformation.scss]: Please use `rem` units instead of percentages.

File: web/themes/custom/subtheme_videogame/src/components/field/_cta.scss [hunk 5]
```diff
@@ -0,0 +1,17 @@
+.cta {
+  &__body {
+    position: relative;
+    height: 40vh;
+    width: 90%;
+    display: table-cell;
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/components/field/_cta.scss]: This looks like an oldschool hack, please use the awesome *Flexbox* if you need to align elements:
https://css-tricks.com/snippets/css/a-guide-to-flexbox
https://css-tricks.com/centering-css-complete-guide

File: web/themes/custom/subtheme_videogame/src/components/field/_cta.scss [hunk 6]
```diff
@@ -0,0 +1,17 @@
+.cta {
+  &__body {
+    position: relative;
+    height: 40vh;
+    width: 90%;
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/components/field/_cta.scss]: I don't like percentages.. You can add margin/padding to control the spacing.

File: web/themes/custom/subtheme_videogame/src/components/field/_cta.scss [hunk 7]
```diff
@@ -0,0 +1,17 @@
+.cta {
+  &__body {
+    position: relative;
+    height: 40vh;
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/components/field/_cta.scss]: You can control the spacing and alignment with margin/padding and flexbox. It's more dynamic then adding fixed values. If you use fixed values, it might not look very good on different resolutions.

File: web/themes/custom/subtheme_videogame/src/components/field/_featuredgames.scss [hunk 8]
```diff
@@ -0,0 +1,34 @@
+.featured-game {
+  &__field-featured-game {
+    background-color: #455A64;
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/components/field/_featuredgames.scss]: as the `#455A64` hex code is used in more places, you can create a variable for this. This is good, because if you ever need to override this value, you'll just need to do it in the `variables.scss` file!

File: web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss [hunk 9]
```diff
@@ -0,0 +1,50 @@
+.genres-to-play {
+  &__body {
+    margin-top: 5%;
+  }
+
+  &__field-genre1, &__field-genre2, &__field-genre3, &__field-genre4, &__field-genre5, &__field-genre6, &__field-genre7 {
+    width: 120px;
+    height: 120px;
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss]: You can control this with an image style.

File: web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    background-color: $orange;
+    font-weight: bold;
+    display: flex;
+    align-items: center;
+    justify-content: center;
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss]: <3

File: web/themes/custom/subtheme_videogame/templates/field/field--block-content--other-games.html.twig [hunk 11]
```diff
@@ -0,0 +1,48 @@
+{#
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/templates/field/field--block-content--other-games.html.twig]: What is the difference between these templates?

File: config/sync/system.site.yml [hunk 12]
```diff
@@ -5,7 +5,7 @@ slogan: ''
 page:
   403: ''
   404: ''
-  front: /node
+  front: /node/10
```
Comment (dzadza) [inline config/sync/system.site.yml:8]: Hmm this is a classical issue. The problem is, that the node ID of the homepage on the live environment may/will differ in most cases. There are workarounds for this. E.g the best is to ignore this config file, whith the [Config Ignore](https://www.drupal.org/project/config_ignore) modole. It's pretty easy to set it up.

**Review by dzadza — CHANGES_REQUESTED** [review]
I'm very impressed with this PR. This is a complex task and you managed to solve it very nice, espacially that you didn't mess up the configs. :heavy_plus_sign: :heavy_plus_sign: However there are some smaller changes that woud be nice to fix. I saw that you duplicated a lot of fields with the same configuration. As I see it, you could have only one `body` and one `image` field in most cases and reuse them. Is there any special reason behind this? It's a better practice to reuse fields when possible, so Drupal won't create separate storage for them.
Also please revert the permission changes for the `web/sites/default/example.settings.local.php` and `web/sites/default/settings.php` files.
At the scss part, my comments apply for the whole PR, so please fix those in every file.

File: web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss (hunk 9 above)
Comment (ilci) [reply web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss]: This is a text field type, so I think I can't give it an image style, that is the reason why I wrote it in the scss file.

**Comment by ilci** [comment]
"Also please revert the permission changes for the web/sites/default/example.settings.local.php and web/sites/default/settings.php files." 
When I type composer install command, I always get some errors, so I type chmod 777 -R web/sites/default, then it runs completely. Maybe that is the reason why the permissions changed. Should I use another command or do something else?

File: web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss (hunk 9 above)
Comment (dzadza) [reply web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss]: Ahh ok, then it's fine.

File: web/themes/custom/subtheme_videogame/src/components/field/_companyinformation.scss [hunk 13]
```diff
@@ -0,0 +1,12 @@
+.company-information {
+  &__body {
+    margin: 4rem 0 4rem 0;
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/components/field/_companyinformation.scss]: This can be simplified to `margin: 4rem 0;`
https://www.w3schools.com/css/css_margin.asp

File: web/themes/custom/subtheme_videogame/src/components/field/_companyinformation.scss [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+    display: flex;
+    align-content: center;
+    justify-content: center;
+
+    p {
+      margin: 0rem 4rem 0rem 4rem;
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/components/field/_companyinformation.scss]: Here as well

File: web/themes/custom/subtheme_videogame/src/components/field/_cta.scss [hunk 15]
```diff
@@ -0,0 +1,21 @@
+.cta {
+  &__body {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    margin-top: 3rem;
+    padding: 3rem 0 3rem 0;
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/components/field/_cta.scss]: and here

File: web/themes/custom/subtheme_videogame/src/components/field/_featuredgames.scss [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+    opacity: 0.8;
+    overflow: hidden;
+    outline: 0.031rem solid $black;
+  }
+}
+
+.games-to-fight-the-day {
+  &__body {
+    background: $sanjuan;
+    padding: 0.3rem 1rem 0.2rem 1rem;
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/components/field/_featuredgames.scss]: `padding: 0.3rem 1rem 0.2rem;` is the same

File: web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+
+.field__item {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  margin: 0 1rem 0.5rem 0;
+  width: 120px;
+  height: 120px;
+
+  &:nth-of-type(1), &:nth-of-type(7) {
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss]: You can define these more dinamically: https://css-tricks.com/useful-nth-child-recipies

File: web/themes/custom/subtheme_videogame/src/components/field/_heroshot.scss [hunk 18]
```diff
@@ -0,0 +1,14 @@
+.heroshot {
+  &__body {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    padding: 8rem 0 8rem 0;
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/components/field/_heroshot.scss]: `8rem 0;`

File: web/themes/custom/subtheme_videogame/src/sass/block/_block.scss [hunk 19]
```diff
@@ -0,0 +1,5 @@
+.block {
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/sass/block/_block.scss]: This is a general change, itt will apply to all the block. Why is this necessary?

**Review by dzadza — CHANGES_REQUESTED** [review]

File: web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss (hunk 17 above)
Comment (ilci) [reply web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss]: I could set the same colors for these ones:
 &:nth-child(6n+1) {
    background-color: rgba($orange, 0.7);
  }  <--- The 1st and the 7th.

  &:nth-child(4n+2) {
    background-color: rgba($orange, 0.8);
  } <--- The 2nd and 6th.

But I can't find with this method only the 3rd and 5th because with "2n+3" the 7th gets the same color. :/

File: web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss [hunk 20]
```diff
@@ -0,0 +1,37 @@
...
+
+  &:nth-child(6n+1) {
+    background-color: rgba($orange, 0.7);
+  }
+
+  &:nth-child(4n+2) {
+    background-color: rgba($orange, 0.8);
+  }
+
+  &:nth-child(3), &:nth-child(5) {
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss]: why not `(6n+3) and (6n+5)` like the previous ones?

File: web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss [hunk 21]
```diff
@@ -0,0 +1,37 @@
...
+    background-color: rgba($orange, 0.9);
+  }
+
+  &:nth-child(4) {
```
Comment (dzadza) [inline web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss]: same here

File: web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss (hunk 20 above)
Comment (dzadza) [reply web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss]: so if there'll ever be a next row, it would look good. :)

File: web/themes/custom/subtheme_videogame/src/sass/block/_block.scss (hunk 19 above)
Comment (dzadza) [reply web/themes/custom/subtheme_videogame/src/sass/block/_block.scss]: So this can be removed now I guess. :)

File: config/sync/field.field.block_content.featured_game.field_background_image.yml [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+required: false
+translatable: true
+default_value: {  }
+default_value_callback: ''
+settings:
+  css_settings:
+    css_selector: ''
+    css_repeat: inherit
+    css_background_size: inherit
+    css_background_position: inherit
```
Comment (dzadza) [inline config/sync/field.field.block_content.featured_game.field_background_image.yml:25]: How is this working? :) Shouldn't you add a selector here?
Also, are the images responsive?
Consider using the following values:
```
css_repeat: no-repeat
css_background_size: cover
css_background_position: center (or according to your needs)
```
https://www.w3schools.com/cssref/pr_background-repeat.asp
https://www.w3schools.com/cssref/css3_pr_background-size.asp
https://www.w3schools.com/cssref/pr_background-position.asp

File: web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss (hunk 17 above)
Comment (dzadza) [reply web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss]: what about 6n+3 and 6n+6? One row contains 6 elements, right?

Comment (dzadza) [reply web/themes/custom/subtheme_videogame/src/components/field/_genrestoplay.scss]: or if there are 7 elements in a row, you can use 7n+1, 7n+2 etc

**Review by dzadza — APPROVED** [review]

**Comment by dzadza** [comment]
> "Also please revert the permission changes for the web/sites/default/example.settings.local.php and web/sites/default/settings.php files."
> When I type composer install command, I always get some errors, so I type chmod 777 -R web/sites/default, then it runs completely. Maybe that is the reason why the permissions changed. Should I use another command or do something else?

**Review by kittiradovics — APPROVED** [review]
